### PR TITLE
feat: validate if the ssh key provided already exists then use the same.

### PR DESCRIPTION
**Pull request by imprateeksh**
### Description
Refer [Issue-355](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/issues/355)

### Types of changes in this PR
Changes are done to ensure if existing ssh key is passed as input which already exists in the specific account/region then no error should be shown.

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [x] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content
  - Compare the ssh key passed as input with the existing keys present in account/region.
  - If present, use the existing ssh key else create a new one.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Comment by imprateeksh** [comment]
Here are the logs of all patterns (with new key and without new key) : 
[roks pattern using existing key](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/files/11190830/apply-roks-with-existing.txt)
[roks pattern using new key](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/files/11190832/apply-roks-with-new.txt)

[mixed pattern using new key](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/files/11190849/apply-mixed-new-key.txt)

(Will update others shortly )

File: patterns/roks/main.tf [hunk 1]
```diff
@@ -15,6 +15,17 @@ provider "ibm" {
 # Landing Zone
 ##############################################################################
 
+data "ibm_is_ssh_keys" "existing_keys" {}
+
+locals {
```
Comment (vburckhardt) [inline patterns/roks/main.tf:20]: Could we avoid the duplication of logic /code across the 3 patterns here. Perhaps put that code in the ssh_key submodule

File: patterns/roks/main.tf [hunk 2]
```diff
@@ -15,6 +15,17 @@ provider "ibm" {
...
+  # compare the remote keys with input variable
+  existing_ssh_key_id = { for key in data.ibm_is_ssh_keys.existing_keys.keys : key.name => key.id if key.public_key == replace(var.ssh_public_key == null ? "" : var.ssh_public_key, "/==.*$/", "==") }
```
Comment (vburckhardt) [inline patterns/roks/main.tf:22]: It'd be nice to add a comment on the need for the replace here.

**Review by vburckhardt — CHANGES_REQUESTED** [review]
Hello @imprateeksh - thanks for the PR. Added 2 comments.

File: patterns/roks/main.tf (hunk 2 above)
Comment (imprateeksh) [reply patterns/roks/main.tf:22]: Sure. I have added the comment.

File: patterns/roks/main.tf (hunk 1 above)
Comment (imprateeksh) [reply patterns/roks/main.tf:20]: @vburckhardt - This requires some time to investigate and put the logic at a central place, for instance passing the id of existing key if logic is added in ssh_key submodule needs more time. 
Can we create a separate issue to track this change and if possible we can proceed with merging the changes?

Comment (imprateeksh) [reply patterns/roks/main.tf:20]: Sure - I have created the [issue-4483](https://github.ibm.com/GoldenEye/issues/issues/4483) to track this change. 
Also, @vburckhardt Is it possible to get this PR merged if the changes looks fine to you?

**Review by vburckhardt — APPROVED** [review]
Ok to address https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/pull/380#discussion_r1162649392 in subsequent iteration.

File: patterns/roks/main.tf (hunk 1 above)
Comment (imprateeksh) [reply patterns/roks/main.tf:20]: This is now taken care with the latest changes. 
PR is raised with the changes and these changes no more exists hence no duplication is present now.

**Pull request merged**